src/keysignature.ts

```diff
@@ -19,115 +19,47 @@ export class KeySignature extends StaveModifier {
   }
 
   protected glyphFontScale: number;
-  protected glyphs: Glyph[];
-  protected xPositions: number[];
+  protected glyphs: Element[];
   protected paddingForced: boolean;
   protected formatted?: boolean;
   protected cancelKeySpec?: string;
   protected accList: { type: string; line: number }[] = [];
   protected keySpec?: string;
   protected alterKeySpec?: string[];
 
-  // Space between natural and following accidental depending
-  // on vertical position
-  static accidentalSpacing: Record<string, { above: number; below: number }> = {
-    '#': {
-      above: 6,
-      below: 4,
-    },
-    b: {
-      above: 4,
-      below: 7,
-    },
-    n: {
-      above: 4,
-      below: 1,
-    },
-    '##': {
-      above: 6,
-      below: 4,
-    },
-    bb: {
-      above: 4,
-      below: 7,
-    },
-    db: {
-      above: 4,
-      below: 7,
-    },
-    d: {
-      above: 4,
-      below: 7,
-    },
-    bbs: {
-      above: 4,
-      below: 7,
-    },
-    '++': {
-      above: 6,
-      below: 4,
-    },
-    '+': {
-      above: 6,
-      below: 4,
-    },
-    '+-': {
-      above: 6,
-      below: 4,
-    },
-    '++-': {
-      above: 6,
-      below: 4,
-    },
-    bs: {
-      above: 4,
-      below: 10,
-    },
-    bss: {
-      above: 4,
-      below: 10,
-    },
-  };
-
   // Create a new Key Signature based on a `keySpec`
   constructor(keySpec: string, cancelKeySpec?: string, alterKeySpec?: string[]) {
     super();
 
     this.setKeySig(keySpec, cancelKeySpec, alterKeySpec);
     this.setPosition(StaveModifierPosition.BEGIN);
-    this.glyphFontScale = Tables.NOTATION_FONT_SCALE;
+    this.glyphFontScale = Tables.lookupMetric('fontSize');
     this.glyphs = [];
-    this.xPositions = []; // relative to this.x
     this.paddingForced = false;
   }
 
   // Add an accidental glyph to the `KeySignature` instance which represents
   // the provided `acc`. If `nextAcc` is also provided, the appropriate
   // spacing will be included in the glyph's position
-  convertToGlyph(acc: { type: string; line: number }, nextAcc: { type: string; line: number }): void {
-    const accGlyphData = Tables.accidentalCodes(acc.type);
-    const glyph = new Glyph(accGlyphData.code, this.glyphFontScale);
+  convertToGlyph(acc: { type: string; line: number }, nextAcc: { type: string; line: number }, stave: Stave): void {
+    const code = Tables.accidentalCodes(acc.type);
+    const glyph = new Element(Category.KeySignature);
+    glyph.setText(code);
+    glyph.measureText();
```

Comment:
Interesting.... the new SMuFL approach seems to make the various subclasses less important. 

Can everything just be an `Element` with a setText(...) as long as we pass in the correct SMuFL code point?

Reply:
Almost everything, or better said, everything that is a SMuFL glyph.